tests/phpunit/tests/block-template-utils.php

```diff
@@ -27,27 +49,170 @@ public static function wpSetUpBeforeClass() {
 		);
 		self::$post = self::factory()->post->create_and_get( $args );
 		wp_set_post_terms( self::$post->ID, get_stylesheet(), 'wp_theme' );
+
+		// Set up template part post.
+		$template_part_args       = array(
+			'post_type'    => 'wp_template_part',
+			'post_name'    => 'my_template_part',
+			'post_title'   => 'My Template Part',
+			'post_content' => 'Content',
+			'post_excerpt' => 'Description of my template part',
+			'tax_input'    => array(
+				'wp_theme'              => array(
+					get_stylesheet(),
+				),
+				'wp_template_part_area' => array(
+					WP_TEMPLATE_PART_AREA_HEADER,
+				),
+			),
+		);
+		self::$template_part_post = self::factory()->post->create_and_get( $template_part_args );
+		wp_set_post_terms( self::$template_part_post->ID, WP_TEMPLATE_PART_AREA_HEADER, 'wp_template_part_area' );
+		wp_set_post_terms( self::$template_part_post->ID, get_stylesheet(), 'wp_theme' );
 	}
 
 	public static function wpTearDownAfterClass() {
 		wp_delete_post( self::$post->ID );
 	}
 
-	function test_build_template_result_from_post() {
-		$template = _build_template_result_from_post(
+	function test_build_block_template_result_from_file() {
+		$template = _build_block_template_result_from_file(
+			array(
+				'slug' => 'single',
+				'path' => __DIR__ . '/../data/templates/template.html',
+			),
+			'wp_template'
+		);
+
+		$this->assertEquals( get_stylesheet() . '//single', $template->id );
+		$this->assertEquals( get_stylesheet(), $template->theme );
+		$this->assertEquals( 'single', $template->slug );
+		$this->assertEquals( 'publish', $template->status );
+		$this->assertEquals( 'theme', $template->source );
+		$this->assertEquals( 'Single Post', $template->title );
+		$this->assertEquals( 'Template used to display a single blog post.', $template->description );
+		$this->assertEquals( 'wp_template', $template->type );
```

Comment:
With PHP getting more and more strict, an effort is ongoing to replace all `assertEquals` with `assertSame`.

```suggestion
		$this->assertSame( get_stylesheet() . '//single', $template->id );
		$this->assertSame( get_stylesheet(), $template->theme );
		$this->assertSame( 'single', $template->slug );
		$this->assertSame( 'publish', $template->status );
		$this->assertSame( 'theme', $template->source );
		$this->assertSame( 'Single Post', $template->title );
		$this->assertSame( 'Template used to display a single blog post.', $template->description );
		$this->assertSame( 'wp_template', $template->type );
```

In the event any of these assertions fails, the rest will not run. To identify which assertion failed, a custom error message can be specified as the 3rd param. These error messages can be added later and are not a blocker for merge.